fuel-merkle/src/common/msb.rs

```diff
@@ -39,12 +39,16 @@ impl<const N: usize> Msb for [u8; N] {
 
     fn common_prefix_count(&self, other: &Self) -> usize {
         let mut count = 0;
-        for i in 0..(N * 8) {
-            let lhs_bit = self.get_bit_at_index_from_msb(i).unwrap();
-            let rhs_bit = other.get_bit_at_index_from_msb(i).unwrap();
-            if lhs_bit == rhs_bit {
-                count += 1;
+        for (byte1, byte2) in self.iter().zip(other.iter()) {
+            let xor = byte1 ^ byte2;
+            if xor == 0 {
+                // The bytes are equal: Increment the prefix count by 8 bits
+                count += 8;
             } else {
+                // The bytes are not equal: Count the number of matching bits
+                // and increment the prefix count by this number
+                let leading_zeros = xor.leading_zeros();
```

Comment:
Many architectures have intrinsics for counting leading zeroes, so this could be even faster than manual bit manipulation techniques.

Reply:
I think this could be further simplified to

```rust
count += (byte1 ^ byte2).leading_zeros();
if byte1 != byte2 {
  break;
}
```

The compiler surely will figure it out.

Reply:
Nice!